assets/app/scripts/controllers/createRoute.js

```diff
@@ -37,66 +37,24 @@ angular.module('openshiftConsole')
         $scope.breadcrumbs[0].title = $filter('displayName')(project);
-        var updatePortOptions = function(service) {
-          if (!service) {
-            return;
-          }
-
-          var unnamedServicePort;
-          if (service.spec.ports.length === 1 && !service.spec.ports[0].name) {
-            unnamedServicePort = true;
-            $scope.alerts['unnamed-service-port'] = {
-              type: "warning",
-              message: "Service " + service.metadata.name + " has a single, unnamed port.",
-              details: "A route cannot specifically target an unnamed service port. " +
-                "If more service ports are added later, the route will also direct traffic to them."
-            };
-          }
-
-          // Only show port options when there is more than one port or when a
-          // single service port has a name. We want to use the service port
-          // name when creating a route. (Port name is required for services
-          // with more than one port.)
-          if (service.spec.ports.length && !unnamedServicePort) {
-            $scope.routing.portOptions = _.map(service.spec.ports, function(portMapping) {
-              return {
-                port: portMapping.name,
-                // \u2192 is a Unicode right arrow.
-                label: portMapping.port + " \u2192 " +
-                       portMapping.targetPort + " (" + portMapping.protocol + ")"
-              };
-            });
-            $scope.routing.targetPort = $scope.routing.portOptions[0];
-          } else {
-            $scope.routing.portOptions = [];
-          }
-        };
 
         var labels = {},
             orderByDisplayName = $filter('orderByDisplayName');
-        if ($scope.serviceName) {
-          DataService.get("services", $scope.serviceName, context).then(function(service) {
-            updatePortOptions(service);
-            labels = angular.copy(service.metadata.labels);
+
+        DataService.list("services", context, function(services) {
+          $scope.services = orderByDisplayName(services.by("metadata.name"));
+          $scope.routing.service = _.find($scope.services, function(service) {
+            return !$scope.serviceName || service.metadata.name === $scope.serviceName;
           });
-        } else {
-          // Prompt the user for the service.
-          DataService.list("services", context, function(services) {
-            $scope.services = orderByDisplayName(services.by("metadata.name"));
-            if (!$scope.routing.service && $scope.services.length) {
-              $scope.routing.service = $scope.services[0];
-            }
-            $scope.$watch('routing.service', function() {
-              updatePortOptions($scope.routing.service);
-              labels = angular.copy($scope.routing.service.metadata.labels);
-            });
+          $scope.$watch('routing.service', function() {
+            labels = angular.copy($scope.routing.service.metadata.labels);
```

Comment:
are we letting you modify the labels today? or just copying them from the service to the route